### PR TITLE
Fixs draft showing when filtering by other statusTypes

### PR DESCRIPTION
**Pull request by orelbn**
*Issue #:*
-  bcgov/entity#16279
-  bcgov/entity#16358

*Description of changes:*
- Only adds Draft registrations when there is no statusTypes filter or filtering by draft
- Updates StatusTypes to fix bugs
- Adds and updates tests

*Before change:*

Screenshot 2023-05-18 082527
Screenshot 2023-05-18 082649

*After change:*

Screenshot 2023-05-18 082854
Screenshot 2023-05-18 082940

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/tests/unit/TableRow.spec.ts [hunk 1]
```diff
@@ -538,6 +538,7 @@ describe('Mhr TableRow tests', () => {
           break
         case MhApiStatusTypes.DRAFT:
           expect(rowData.at(3).text()).toContain(MhUIStatusTypes.DRAFT)
+          expect(rowData.at(rowData.length - 1).text()).toContain('Edit')
```
Comment (orelbn) [inline ppr-ui/tests/unit/TableRow.spec.ts:541]: I don't believe that at this point we are considering authorization when showing the Edit button, might need to update that.

File: ppr-ui/tests/unit/TombstoneDischarge.spec.ts [hunk 2]
```diff
@@ -58,28 +58,35 @@ describe('Tombstone component', () => {
   }
   const registrationType = mockedSelectSecurityAgreement()
 
-  beforeEach(async () => {
-    // mock the window.location.assign function
-    delete window.location
-    window.location = { assign: jest.fn() } as any
-
+  beforeAll(async () => {
```
Comment (orelbn) [inline ppr-ui/tests/unit/TombstoneDischarge.spec.ts:61]: We don't change the data in the store in these tests, so we can just set it once.

File: ppr-ui/tests/unit/TombstoneDischarge.spec.ts [hunk 3]
```diff
@@ -58,28 +58,35 @@ describe('Tombstone component', () => {
...
+    delete window.location
+    window.location = { assign: jest.fn() } as any
+  })
+
   afterEach(() => {
     window.location.assign = assign
     wrapper.destroy()
   })
 
+  afterAll(async () => {
```
Comment (orelbn) [inline ppr-ui/tests/unit/TombstoneDischarge.spec.ts:80]: Clean up, so it doesn't interfere with the added MHR test

File: ppr-ui/tests/unit/TombstoneDischarge.spec.ts [hunk 4]
```diff
@@ -58,28 +58,35 @@ describe('Tombstone component', () => {
...
+    await store.dispatch('setRegistrationType', null)
+    await store.dispatch('setRegistrationNumber', null)
+    await store.dispatch('setRegistrationCreationDate', null)
+    await store.dispatch('setRegistrationExpiryDate', null)
+  })
+
   it('renders Tombstone component properly for Total Discharge', async () => {
-    await store.dispatch('setMhrInformation', mockedMhrInformation)
```
Comment (orelbn) [inline ppr-ui/tests/unit/TombstoneDischarge.spec.ts:79]: I don't believe the deleted lines do anything in the test, specifically cause the code does not call Vue.nextTick, and the tests seem to be meant for PPR.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1342-ovl766n1.web.app

**Review by cameron-eyds — APPROVED** [review]
Looks good, nice work 👍

**Review by dimak1 — APPROVED** [review]
Nice!

**Review by doug-lovett — APPROVED** [review]
Looks great, very nice.

**Pull request merged**